contributions/catalog/controllers/auth.py

```diff
@@ -0,0 +1,26 @@
+#  Copyright 2020 Board of Trustees of the University of Illinois.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import functools
+from flask import (
+    Blueprint, flash, g, redirect, render_template, request, session, url_for, current_app
+)
+
+def login_required(view):
```

Comment:
I am not really sure and I am not expert of it but is it better to make this as class? Or is it better to have def inside the def? I am just asking since I don't know too much about this.

Reply:
I am not sure about this - let us test this bit more and I can make the change if we feel needed.